pkg/registry/apis/provisioning/test.go

```diff
@@ -53,34 +54,37 @@ func (s *testConnector) Connect(ctx context.Context, name string, opts runtime.O
 		}
 		var repo repository.Repository
 		if len(body) > 0 {
-			cfg := &provisioning.Repository{}
-			err = json.Unmarshal(body, cfg)
+			var cfg provisioning.Repository
+			err = json.Unmarshal(body, &cfg)
 			if err != nil {
 				responder.Error(err)
 				return
 			}
 
-			// Create a temporary repository
-			tmp, err := s.getter.AsRepository(ctx, cfg)
-			if err != nil {
-				responder.Error(err)
-				return
-			}
-
-			if name != "new" {
-				repo, err = s.getter.AsRepository(ctx, cfg)
+			// In case the body is an empty object
+			if !reflect.ValueOf(cfg).IsZero() {
```

Comment:
I'd definitely prefer `if cfg != provisioning.Repository{}` or similar with an intentional zero value

Reply:
@Proximyst  try that syntax locally. It doesn't work because of the curly braces for the empty object collides with the ones in the if.